controller/src/main/java/org/jboss/as/controller/operations/global/ReadFeatureHandler.java

```diff
@@ -0,0 +1,1179 @@
+            String paramName = attr.getName();
+            requestProps.add(new Property(paramName, new ModelNode()));
+            if(idParams.contains(paramName)) {
+                paramName += "-feature";
+                if(opParamMapping.isEmpty()) {
+                    opParamMapping = Collections.singletonMap(attr.getName(), paramName);
+                } else {
+                    if(opParamMapping.size() == 1) {
+                        opParamMapping = new HashMap<>(opParamMapping);
+                    }
+                    opParamMapping.put(attr.getName(), paramName);
+                }
+            }
+            param.get(ModelDescriptionConstants.NAME).set(paramName);
+            if(!attr.isRequired()) {
+                param.get(NILLABLE).set(true);
+            }
+            if (objAttDef.getDefaultValue() != null && objAttDef.getDefaultValue().isDefined()) {
+                param.get(ModelDescriptionConstants.DEFAULT).set(objAttDef.getDefaultValue());
+            }
+            if(attr.getType() == ModelType.LIST) {
+                param.get(TYPE).set("List<String>");
+            }
+            params.add(param);
+        }
+        addOpParam(annotation, requestProps, opParamMapping);
+
+        parentFeature.get(CHILDREN).get(specName).set(attFeature);
+    }
+
+    private void processListAttribute(final ModelNode parentFeature, final ImmutableManagementResourceRegistration registration, ObjectListAttributeDefinition objAttDef) {
+        //System.out.println("List attr " + registration.getPathAddress().toCLIStyleString() + " " + objAttDef.getName());
+        final ModelNode parentSpecName = parentFeature.require(NAME);
+        ModelNode attFeature = new ModelNode();
+        final String specName = parentSpecName.asString() + "." + objAttDef.getName();
+        attFeature.get(NAME).set(specName);
+        ModelNode annotation = attFeature.get(ANNOTATION);
+        annotation.get(ModelDescriptionConstants.NAME).set("list-add");
+        annotation.get("complex-attribute").set(objAttDef.getName());
+        if (parentFeature.hasDefined(ANNOTATION)) {
+            annotation.get(ADDRESS_PARAMETERS)
+                    .set(parentFeature.require(ANNOTATION).require(ADDRESS_PARAMETERS));
+            if (parentFeature.require(ANNOTATION).hasDefined(ADDRESS_PARAMETERS_MAPPING)) {
+                annotation.get(ADDRESS_PARAMETERS)
+                        .set(parentFeature.require(ANNOTATION).require(ADDRESS_PARAMETERS_MAPPING));
+            }
+        } else {
+            addParams(attFeature, registration.getPathAddress(), new ModelNode().setEmptyList());
+        }
+
+        ModelNode refs = attFeature.get(REFERENCES).setEmptyList();
+        ModelNode ref = new ModelNode();
+        ref.get(FEATURE).set(parentSpecName);
+        refs.add(ref);
+
+        ModelNode params = attFeature.get(PARAMETERS).setEmptyList();
+        Set<String> idParams = new HashSet<>();
```

Comment:
This idParams set is never populated, which mean the if check at L547 will never be true.

Of all these IDE warns I'm reporting this is the one that's not just housekeeping but may indicate a problem.